pkg/operator/ceph/object/user/controller.go

```diff
@@ -408,17 +409,38 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 			logCreateOrUpdate = fmt.Sprintf("created ceph object user %q", u.Name)
 		} else if strings.Contains(err.Error(), "InvalidAccessKeyId") {
 			// In case of an Invalid Access Key, delete the `rgw-admin-ops-user` and restart the operator.
-			// This is a temporary fix until https://bugzilla.redhat.com/show_bug.cgi?id=2373031 is resolved.
+			// This is a temporary workaround until https://bugzilla.redhat.com/show_bug.cgi?id=2373031 is resolved.
 			logger.Errorf("failed to get Ceph Object user %q. Error: %v", u.Name, err)
-			logger.Infof("deleting the admin user %q", cephObject.RGWAdminOpsUserSecretName)
-			_, err := cephObject.DeleteUser(&r.objContext.Context, cephObject.RGWAdminOpsUserSecretName)
-			if err != nil {
-				return errors.Wrapf(err, "failed to delete admin user %q with InvalidAccessKeyID",
-					cephObject.RGWAdminOpsUserSecretName)
+			waSuccess := false
+			// user deletion can hang due to the same underlying reasons that create fails.
+			// in testing, radosgw-admin seemed to respond more often with ~30 second periodicity,
+			// so retry the deletion workaround with 35 second cooldown hoping for better chance of success.
+			for i := range 4 {
+				if i > 0 {
+					logger.Infof("waiting 35 seconds before attempting to delete admin user %q again", cephObject.RGWAdminOpsUserSecretName)
+					time.Sleep(35 * time.Second)
```

Comment:
Where did 35s come from? Just an arbitrary reasonable value?

Reply:
Since we were seeing the command seem to work well at around 30 seconds, it seemed reasonable to give just a little extra cooldown time beyond that. I guess I could have just as easily used 33 seconds, but 35 was nice and round